src/components/icons/LottieAnimation.tsx

```diff
@@ -1,38 +1,146 @@
+import _ from 'lodash'
 import Player, { LottieRefCurrentProps } from 'lottie-react'
-import React, { useEffect, useRef } from 'react'
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import LottieData, { RGBA, ShapeItem, ShapeLayer } from '../../@types/LottieTypes'
 
+/** The props for the LottieAnimation component. */
 interface LottieAnimationProps {
-  animationData: any
+  animationData: LottieData | null
   speed?: number
+  color?: string
   onComplete?: () => void
   style?: React.CSSProperties
 }
 
+/**
+ * Converts hex color to RGBA.
+ *
+ * @param hex - The hex color code.
+ * @returns The RGBA values.
+ */
+const hexToRGBA = (hex: string): RGBA => {
```

Comment:
Would it make sense to also use `hexToRgb` from `@mui/material`? It does support rgba, but I'm not sure if it would be worth extracting the values from the string if we need an array of numbers:

```
> hexToRgb('#ffffffaa')
'rgba(255, 255, 255, 0.667)'
```

Reply:
Parsing hex colors directly is generally simpler because hex values are compact, predictable, and easier to process programmatically.

Reply:
If we are using `rgbToHex` from the `@mui/material` library, we should also use `hexToRgb` from the library for consistency. It doesn't matter how simple the implementation is when it has already been implemented for us. 

If implementing it ourselves meant we could remove `@mui/material` as a dependency, that would be a compelling case. As it stands there does not seem to be anything to gain.